actors/miner/tests/prove_commit_sector_3_test.rs

```diff
@@ -489,6 +489,418 @@ fn rejected_notification_dropped() {
+    let cfg = ProveCommitSectors3Config {
+        validation_failure: vec![0, 1], // The first two are expired
+        ..Default::default()
+    };
+
+    // With require_activation_success=false, partial success is allowed
+    let (result, claims, notifications) =
+        h.prove_commit_sectors3(&rt, &manifests, false, false, true, cfg).unwrap();
+
+    // First two fail, rest succeed
+    assert_commit_result(
+        &[
+            ExitCode::USR_ILLEGAL_ARGUMENT,
+            ExitCode::USR_ILLEGAL_ARGUMENT,
+            ExitCode::OK,
+            ExitCode::OK,
+            ExitCode::OK,
+            ExitCode::OK,
+        ],
+        &result,
+    );
+
+    // Verify only successful sectors were committed
+    let st = h.get_state(&rt);
+    assert!(st.get_sector(&rt.store, snos[0]).unwrap().is_none());
+    assert!(st.get_sector(&rt.store, snos[1]).unwrap().is_none());
+
+    for sno in &snos[2..] {
+        let sector = h.get_sector(&rt, *sno);
+        assert_eq!(sector.sector_number, *sno);
+        verify_weights(&rt, &h, *sno, 0, piece_size);
+    }
+
+    // Only successful sectors have claims
+    assert_eq!(4, claims.len()); // Only 4 successful sectors
+    let successful_claims = claims.iter().filter(|c| !c.claims.is_empty()).count();
+    assert_eq!(4, successful_claims); // All 4 successful sectors have actual claims
+
+    // Only successful sectors have notifications
+    assert_eq!(4, notifications.len());
+
+    h.check_state(&rt);
+}
+
+#[test]
+fn aggregate_proof_max_proof_size() {
+    // Test aggregate proof at maximum allowed proof size
+    let (h, mut rt) = setup_basic();
+    let piece_size = h.sector_size as u64;
+    let sector_count =
+        fil_actors_runtime::runtime::policy_constants::MIN_AGGREGATED_SECTORS as usize;
+
+    let piece_config = [piece_size];
+    let piece_configs: Vec<&[u64]> = (0..sector_count).map(|_| &piece_config[..]).collect();
+    let precommits = precommit_sectors(&mut rt, &h, &piece_configs);
+    let snos: Vec<SectorNumber> =
+        precommits.iter().map(|pci: &SectorPreCommitInfo| pci.sector_number).collect();
+
+    let manifests: Vec<_> =
+        snos.iter().map(|sno| make_activation_manifest(*sno, &[(piece_size, 0, 0, 0)])).collect();
+
+    // Create a proof at exactly the maximum size limit
+    // Note: The max proof size is quite large, we'll use a smaller proof for the test
+    // that still demonstrates the validation works
+    let cfg = ProveCommitSectors3Config::default();
+
+    let (result, _, _) = h.prove_commit_sectors3(&rt, &manifests, true, false, true, cfg).unwrap();
+
+    assert_commit_result(&vec![ExitCode::OK; sector_count], &result);
+
+    // Verify all sectors were committed
+    for sno in snos {
+        let sector = h.get_sector(&rt, sno);
+        assert_eq!(sector.sector_number, sno);
+        verify_weights(&rt, &h, sno, piece_size, 0);
+    }
+
+    h.check_state(&rt);
+}
+
+#[test]
+fn aggregate_proof_with_all_features() {
```

Comment:
I think this is redundant enough that we can just remove this one.

Reply:
Removed